rfd/0088-passwordless-windows.md

```diff
@@ -0,0 +1,243 @@
+---
+authors: Tobiasz Heller (tobiasz.heller@goteleport.com)
+state: draft
+---
+
+# RFD 0088 - Passwordless for windows CLI
+
+## Required Approvers
+
+* Engineering @rosstimothy && @codingllama
+* Security @reed
+* Product: (@xinding33 || @klizhentas )
+
+## What
+
+Passwordless features for native windows CLIs, aka Windows Hello and FIDO
+support for CLI/`tsh`.
+
+This is a part of the [Passwordless RFD][passwordless rfd].
+
+Passwordless for Windows is available as a preview in Teleport 11.
+
+## Why
+
+Native, non-browser windows clients lack support for Windows Hello or FIDO
+devices. This RFD explores how we can achieve that support for `tsh` in a
+secure way.
+
+## Details
+
+Windows Hello and FIDO devices support is implemented via win32 API which is
+installed on Windows 10 19H1 or later.
+More details about API can be found in [API spec](https://github.com/microsoft/webauthn)
+or [documentation][webauthn API windows docs].
+
+```mermaid
+flowchart TB
+  WRP[WebAuthn RP]
+  subgraph CD[Client Device]
+    BC[Browser Client]
+    NC[Native Client]
+    subgraph WADLL[WEBAUTHN.dll]
+        PA[Platorm Authenticator]
+        CPA[Cross-platform Authenticator]
+    end
+  end
+  subgraph RA[Roaming Authenticator]
+    CTAP2[CTAP2 API]
+  end
+
+
+  WRP --> CD
+  BC --> WADLL
+  NC --> WADLL
+
+  CPA -. USB/NFC/BLE .-> CTAP2
+```
+
+`Webauthn.dll` is designed to work with Webauthn protocol, that's why
+integration with it results moslty in passing parameters from and back to
+server.
+
+**Important:** Windows system without `webauthn.dll` (older than Windows 10 19H1)
+won't be supported.
+
+If `webauthn.dll` is not found, `tsh` should disable Windows Hello or FIDO
+devices support. DLL will be loaded directly from system.
+Microsoft provides 4 versions of webauthn API and all should be supported.
+
+Interactions (touch messages and pin inputs) with Windows Hello or FIDO devices
+is done via Windows UI.
+
+Windows Hello works with both Microsoft online and offline accounts.
+
+### Registration
```

Comment:
Let's mention attestation here too - do we support anything other than self-attestation for Hello?

Reply:
I have added it here: https://github.com/gravitational/teleport/blob/8f5c6e89bf9eb6eed1e29701ca617836c59c2af7/rfd/0088-passwordless-windows.md#L100-L101
Let me know if this is enough or not.

There is also that issue that with older TPM, in attestation can be used SHA-1. Should we mention it in RFD and try to reject that attestation?

Reply:
Thanks  👍 

Re SHA-1: maybe mention in the RFD, but I wouldn't forbid it if that's the only thing they are capable of.

Reply:
added here https://github.com/gravitational/teleport/blob/823fe8dce07ad10cc3f46121df9cc44d364455e4/rfd/0088-passwordless-windows.md#L101-L103